### DISCUSSION TITLE
Ability to specify association preloading when populating `belongs_to` association selection.

### DISCUSSION THREAD

**Discussion by MrJoy** [avo-hq/avo#805] [Feature requests]
I use a custom `display_name` method for a model (`TransformersToyline`).  It makes use of a certain association (`TransformersVendor`) in synthesizing the final value.  When I'm editing an item that `belongs_to` `TransformersToyline`, the select list is populated using `display_name`, but the `includes` configuration from `TransformersToylineResource` is not used.  Of course, it would be imprudent to use this in most cases, but having the ability to separately specify the association preloading behavior for the populating-a-dropdown list would be very useful in these cases to avoid N+1 situations.

**Comment by adrianthedev** [comment]
Is that something you can do with rails associations?

I mean, this is not valid code.

```ruby
TransformersToyline.find(1).includes(:something).vendor
```

What you could do to avoid n+1 is to make that field a [`searchable` field](https://docs.avohq.io/2.0/associations.html#searchable-belongs-to).

**Comment by MrJoy** [comment]
It seems like what's happening under the hood is that it's getting `TransformersToyline.all`, to populate the `<select>`, and while iterating it's calling `display_name` to get the label value, which then references the `vendor` association.

Thanks for the `searchable` tip.  I was able to toss in a `.includes` and get rid of the N+1 that way.  Although, initially I didn't read the part about needing `self.search_query` set on the target resource and was very confused by the lack of any results.  An error message might be helpful for unobservant folks like myself.

**Reply by adrianthedev** [reply]
Got it! I surfaced that a bit better. Thanks!

<img width="837" alt="CleanShot 2022-04-12 at 10 31 53@2x" src="https://user-images.githubusercontent.com/1334409/162905586-60dea2d4-e0da-48c8-a350-92f5b9ed5162.png">
